lib/ansible/cli/doc.py

```diff
@@ -134,7 +134,7 @@ def _load_role_data(self, root, files, role_name, collection):
                 data = yaml.load(trust_as_template(f), Loader=AnsibleLoader)
                 if data is None:
                     data = {}
-        except (IOError, OSError) as ex:
+        except OSError as ex:
             raise AnsibleParserError(f"Could not read the role {role_name!r} (at {path}).") from ex
```

Comment:
Use `!r` so strings are properly quoted:

```suggestion
            raise AnsibleParserError(f"Could not read the role {role_name!r} at {path!r}.") from ex
```